docs/DeveloperGuide.adoc

```diff
@@ -137,256 +167,944 @@ image:BetterModelClassDiagram.png[]
 .Structure of the Storage Component
 image::StorageClassDiagram.png[]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/storage/Storage.java[`Storage.java`]
+*API* : link:{repoURL}/src/main/java/thrift/storage/Storage.java[`Storage.java`]
 
 The `Storage` component,
 
-* can save `UserPref` objects in json format and read it back.
-* can save the Address Book data in json format and read it back.
+* can save `**UserPref**` objects in json format and read it back.
+* can save the `**THRIFT**` data in json format and read it back:
+** `**JsonAdaptedTransaction**` stores `**Transaction**` objects in `JSON` format.
+** `**JsonAdaptedTag**` stores `**Tag**` objects in `JSON` format, nested under `**JsonAdaptedTransaction**`.
+** `**JsonAdaptedBudget**` stores `**Budget**` objects in `JSON` format.
 
 [[Design-Commons]]
 === Common classes
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+Classes used by multiple components are in the `**thrift.commons**` package.
 
 == Implementation
 
 This section describes some noteworthy details on how certain features are implemented.
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+// tag::budget[]
+=== Adding transactions
+==== Implementation
+The **add_expense**/**add_income** feature is implemented through polymorphism. `**Expense**` and `**Income**` are both subclasses of the abstract class `Transaction`.
```

Comment:
Good effort to differentiate the different elements of the program through formatting! That said, I think you missed a bold on **`Transaction`**.